otelcol/collector.go

```diff
@@ -105,8 +112,14 @@ type Collector struct {
 
 // NewCollector creates and returns a new instance of Collector.
 func NewCollector(set CollectorSettings) (*Collector, error) {
```

Comment:
During the interim period where one can provide both a `ConfigProvider` and a `ConfigProviderSettings`, should we check that if `configProvider` is not nil, then `ConfigProviderSettings` has the zero value for that type and error otherwise?

Reply:
We should do that, good catch. I added some validation to check this.